test/unit/account_info/test_account_info.py

```diff
@@ -414,30 +415,24 @@ def _make_sqlite_account_info(self, env=None, last_upgrade_to_run=None):
                 last_upgrade_to_run=last_upgrade_to_run,
             )
 
-    def test_uses_default(self):
-        account_info = self._make_sqlite_account_info(
-            env={
-                'HOME': self.test_home,
-                'USERPROFILE': self.test_home,
-            }
-        )
-        actual_path = os.path.abspath(account_info.filename)
-        assert os.path.join(self.test_home, '.b2_account_info') == actual_path
-
     def test_uses_xdg_config_home(self, apiver):
+        is_xdg_os = bool(SqliteAccountInfo.get_xdg_config_path())
         with WindowsSafeTempDir() as d:
             account_info = self._make_sqlite_account_info(
                 env={
                     'HOME': self.test_home,
                     'USERPROFILE': self.test_home,
-                    XDG_CONFIG_HOME_ENV_VAR: d,
+                    # pass the env. variable on XDG-like OS only
+                    XDG_CONFIG_HOME_ENV_VAR: is_xdg_os and d,
```

Comment:
this will set `XDG_CONFIG_HOME_ENV_VAR` to `False`, while it doesn't change the outcome of the code, 
this test already relies to much on internal behavior, so lets fix at least this and only add `XDG_CONFIG_HOME_ENV_VAR` when needed to env